File: SDKs/amazonaws.com/clouddirectory/2016-05-10/java/lib/src/main/java/openapisdk/models/shared/BatchGetObjectInformation.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * BatchGetObjectInformation
 * Retrieves metadata about an object inside a <a>BatchRead</a> operation. For more information, see <a>GetObjectInformation</a> and <a>BatchReadRequest$Operations</a>.
**/
public class BatchGetObjectInformation {
    @JsonProperty("ObjectReference")
    public ObjectReference objectReference;
    public BatchGetObjectInformation withObjectReference(ObjectReference objectReference) {
        this.objectReference = objectReference;
        return this;
    }
}

File: SDKs/amazonaws.com/clouddirectory/2017-01-11/python/src/sdk/models/shared/batchgetobjectinformation.py
import dataclasses
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import objectreference as shared_objectreference


@dataclass_json
@dataclasses.dataclass
class BatchGetObjectInformation:
    r"""BatchGetObjectInformation
    Retrieves metadata about an object inside a <a>BatchRead</a> operation. For more information, see <a>GetObjectInformation</a> and <a>BatchReadRequest$Operations</a>.
    """
    
    object_reference: shared_objectreference.ObjectReference = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('ObjectReference') }})
    
